### DISCUSSION TITLE
Unable to create a job in a child application

### DISCUSSION THREAD

**Discussion by ohdearaudrey** [argoproj/argo-cd#9816] [Q&A] [answered]
Using the app of apps model i have an Application that points to a directory with 3 apps, which point to three deployments each. I want to run a postsync job in the application that contains my deployments however the job never shows up in the UI. If i move the job file up a level to where the parent app is the job shows up, and can run. 

I know this is possible because i downloaded a demo of app of apps and i can see a job in their child app but for some reason mine just does not show up. 

any suggestions would help me out a lot. Been banging my head against this for a while now.

**Comment by ishitasequeira** [comment]
Is the Sync successful for the child app you have attached job to?
If yes, maybe try adding some sleep to confirm if the post sync job does come up on the UI or not.

**Comment by ohdearaudrey** [comment]
Yes the sync works for all apps and everything is healthy, and it does the deployments as expected. Sleep did not fix it but the job does not show up even if i change to presync, postsync or any other option etc. Also i do not have job deletion on so i should see the completed container on the cluster if it worked. 

Here is a better illustration of my issue. main-app points to the apps folder, which then starts the frontend-app application which points to the manifests folder which has my job and deployments in it. At this point i would expect the frontend-app application to have a job inside of it but it does not, only a deployment. Then if i put the job-file into the apps folder where main-app is pointed to i see the job show up under main-app alongside frontend-app

argo/
├── apps/
│   └── frontend-app.yaml
├── manifests/
│   ├── job-file.yaml
│   └── deployment-file.yaml
└── main-app.yaml

**Comment by ohdearaudrey** [comment, marked as answer]
I solved my own issue. Kustomize does not support generateName so if like me you were following the Argo Docs and had a job file using generateName in a directory that has a kustomization file in it it makes the job not work with no error. In my case moving the job file to a directory without a kustomization file made it show up.
